ClientUpdater/Updater.cs

```diff
@@ -15,1140 +15,1152 @@
+        {
+            DefaultVersionPolicy = HttpVersionPolicy.RequestVersionOrHigher
+        };
+    }
 
-            GamePath = gamePath;
-            ResourcePath = resourcePath;
-            settingsINI = new IniFile(SafePath.CombineFilePath(GamePath, settingsIniName));
-            LocalGame = localGame;
-            CallingExecutableFileName = callingExecutableFileName;
+    /// <summary>
+    /// Initializes the updater.
+    /// </summary>
+    /// <param name="gamePath">Path of the root client / game folder.</param>
+    /// <param name="resourcePath">Path of the resource folder of client / game.</param>
+    /// <param name="settingsIniName">Client settings INI filename.</param>
+    /// <param name="localGame">Local game ID of the current game.</param>
+    /// <param name="callingExecutableFileName">File name of the calling executable.</param>
+    public static void Initialize(string gamePath, string resourcePath, string settingsIniName, string localGame, string callingExecutableFileName)
+    {
+        Logger.Log("Updater: Initializing updater.");
 
-            ReadUpdaterConfig();
+        GamePath = gamePath;
+        ResourcePath = resourcePath;
+        settingsINI = new(SafePath.CombineFilePath(GamePath, settingsIniName));
+        LocalGame = localGame;
+        CallingExecutableFileName = callingExecutableFileName;
 
-            Logger.Log("Updater: Update mirror count: " + updateMirrors.Count);
-            Logger.Log("Updater: Running from: " + CallingExecutableFileName);
-            List<UpdateMirror> list = new List<UpdateMirror>();
-            List<string> sectionKeys = settingsINI.GetSectionKeys("DownloadMirrors");
+        ReadUpdaterConfig();
 
-            if (sectionKeys != null)
-            {
-                foreach (string str in sectionKeys)
-                {
-                    string value = settingsINI.GetStringValue("DownloadMirrors", str, string.Empty);
-                    UpdateMirror item = updateMirrors.Find(um => um.Name == value);
-                    if ((item != null) && !list.Contains(item))
-                    {
-                        list.Add(item);
-                    }
-                }
-            }
-            foreach (UpdateMirror mirror2 in updateMirrors)
+        Logger.Log("Updater: Update mirror count: " + updateMirrors.Count);
+        Logger.Log("Updater: Running from: " + CallingExecutableFileName);
+        var list = new List<UpdateMirror>();
+        List<string> sectionKeys = settingsINI.GetSectionKeys("DownloadMirrors");
+
+        if (sectionKeys != null)
+        {
+            foreach (string str in sectionKeys)
             {
-                if (!list.Contains(mirror2))
+                string value = settingsINI.GetStringValue("DownloadMirrors", str, string.Empty);
+                UpdateMirror item = updateMirrors.Find(um => um.Name == value);
+                if (item != null && !list.Contains(item))
```

Comment:
```suggestion

                if (item != null && !list.Contains(item))
```
Also maybe omit the braces below